xarray/tests/test_dataset.py

```diff
@@ -4492,3 +4492,38 @@ def test_raise_no_warning_for_nan_in_binary_ops():
     with pytest.warns(None) as record:
         Dataset(data_vars={'x': ('y', [1, 2, np.NaN])}) > 0
     assert len(record) == 0
+
+
+@pytest.mark.parametrize('dask', [True, False])
+@pytest.mark.parametrize('edge_order', [1, 2])
+def test_gradient(dask, edge_order):
```

Comment:
can you add a test for a `datetime` co-ordinate also? The `gradient` in `npcompat` seems to support it.

Reply:
It looks np.gradient does not support `datetime`. Maybe we need a unit that should be used for compute the gradient along `datetime`, such as `ns` or `date`. 
Do you have any idea about API for this?

Reply:
Oh you're right; I misunderstood the code.

How about

```
da = xr.DataArray([1., 2.], dims='time', coords={'time': pd.date_range('2001-01-01', '2001-01-02')})

da.differentiate(coord='time', astype='D')
```

which would be equivalent to
```
np.diff(da.values) / (np.diff(da.time).astype('timedelta64[D]').astype('float64'))
```

`astype` isn't the best kwarg name though. `freq` or `resolution` don't seem correct either.

Reply:
In the long term, hopefully we'll solve this by getting real unit support working, but a `time_unit` argument seems like a good choice for now

Reply:
Numpy seems not to define the terminology for this unit, but *datetime_unit* is being used [here](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.datetime_data.html). I wonder if we should use it here.
Is it a little long?